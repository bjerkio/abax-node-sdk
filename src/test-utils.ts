import { MockAgent, type Interceptable, setGlobalDispatcher } from 'undici';
import { AbaxClient } from './abax-client';

export function initialiseClientAndMockPool(): {
  client: AbaxClient;
  mockPool: Interceptable;
} {
  const mockAgent = new MockAgent();
  mockAgent.disableNetConnect();
  setGlobalDispatcher(mockAgent);

  const mockPool = mockAgent.get('https://api.abax.cloud');

  const client = new AbaxClient({
    baseUrl: 'production',
    apiKey: 'kanonball!',
  });

  return { client, mockPool };
}
