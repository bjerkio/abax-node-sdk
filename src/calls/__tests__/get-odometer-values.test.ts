import { describe, expect, it } from 'vitest';
import { initialiseClientAndMockPool } from '../../test-utils';

describe('get odometer values', () => {
  it('should get odometer values', async () => {
    const { client, mockPool } = initialiseClientAndMockPool();

    mockPool
      .intercept({
        path: '/v1/trips/odometerReadings?trip_ids=871c120e54cb4c58b4fb5ee457bb968a&trip_ids=0e2d0a84463e411dbedab16560b60430&trip_ids=129769b3470354568ae6995e7b3c70f5',
        method: 'GET',
      })
      .reply(200, {
        items: [
          {
            trip_id: '871c120e54cb4c58b4fb5ee457bb968a',
            trip_start: {
              value: 100000000,
              timestamp: '2019-01-10T10:00:00.000+00:00',
            },
            trip_finish: {
              value: 100134000,
              timestamp: '2019-01-10T11:33:12.000+00:00',
            },
          },
          {
            trip_id: '0e2d0a84463e411dbedab16560b60430',
            trip_start: {
              value: 83000,
              timestamp: '2019-01-10T10:00:00.000+00:00',
            },
            trip_finish: {
              value: 89000,
              timestamp: '2019-01-10T11:33:12.000+00:00',
            },
          },
          {
            trip_id: '129769b3470354568ae6995e7b3c70f5',
          },
        ],
      })
      .times(1);

    const odometerValues = await client.getOdometerValuesOfTrips({
      query: {
        tripIds: [
          '871c120e54cb4c58b4fb5ee457bb968a',
          '0e2d0a84463e411dbedab16560b60430',
          '129769b3470354568ae6995e7b3c70f5',
        ],
      },
    });
    expect(odometerValues).toMatchSnapshot();
  });
});
