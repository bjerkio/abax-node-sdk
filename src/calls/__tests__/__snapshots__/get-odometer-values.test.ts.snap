// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`get odometer values > should get odometer values 1`] = `
{
  "items": [
    {
      "tripFinish": {
        "timestamp": "2019-01-10T11:33:12.000+00:00",
        "value": 100134000,
      },
      "tripId": "871c120e54cb4c58b4fb5ee457bb968a",
      "tripStart": {
        "timestamp": "2019-01-10T10:00:00.000+00:00",
        "value": 100000000,
      },
    },
    {
      "tripFinish": {
        "timestamp": "2019-01-10T11:33:12.000+00:00",
        "value": 89000,
      },
      "tripId": "0e2d0a84463e411dbedab16560b60430",
      "tripStart": {
        "timestamp": "2019-01-10T10:00:00.000+00:00",
        "value": 83000,
      },
    },
    {
      "tripFinish": undefined,
      "tripId": "129769b3470354568ae6995e7b3c70f5",
      "tripStart": undefined,
    },
  ],
}
`;
